### PR TITLE
API does not return JSON systematically - fix #10

### LINKED ISSUE #10: Exception if API endpoint does not return JSON
Hi,

I saw that you always use json_decode on get api response : https://github.com/Ubiquiti-App/UCRM-Plugin-SDK/blob/685cec5a4e5d3127b168da3f64f96418d5c67900/src/UcrmPluginSdk/Service/UcrmApi.php#L113-L124
But there is a problem : all endpoints does not return a json systematically. 

For example, this one returns a pdf file : https://ucrm.docs.apiary.io/#reference/invoices/invoicesidpdf/get

Call this endpoint causes an exception even if the response is correct.

`Fatal error: Uncaught Ubnt\UcrmPluginSdk\Exception\JsonException: Syntax error in /data/ucrm/data/plugins/myplugin/vendor/ubnt/ucrm-plugin-sdk/src/UcrmPluginSdk/Util/Json.php:45`

Maybe you can change your code to something like this : 
```php
    public function get(string $endpoint, array $query = [])
    {
        $response = $this->request(
            'GET',
            $endpoint,
            [
                'query' => $query,
            ]
        );
        if(current($response->getHeader('Content-Type')) === 'application/json'){
            return Json::decode((string) $response->getBody());
        }

        return (string) $response->getBody();
    }
```

### PR DESCRIPTION
**Pull request by ajoah**

### CODE REVIEW

File: src/UcrmPluginSdk/Service/UcrmApi.php [hunk 1]
```diff
@@ -119,8 +119,12 @@ public function get(string $endpoint, array $query = []): array
                 'query' => $query,
             ]
         );
+
```
Comment (keksa) [inline src/UcrmPluginSdk/Service/UcrmApi.php]:
```suggestion

```

**Review by keksa — CHANGES_REQUESTED** [review]

File: src/UcrmPluginSdk/Service/UcrmApi.php [hunk 2]
```diff
@@ -105,12 +105,12 @@ public static function create(?string $pluginRootPath = null): self
      *
      * @param mixed[] $query
      *
-     * @return mixed[]
+     * @return array|string
```
Comment (keksa) [inline src/UcrmPluginSdk/Service/UcrmApi.php]: Sorry, the cs check is sometimes too hard on us. :)

```suggestion
     * @return mixed[]|string
```

**Review by keksa — CHANGES_REQUESTED** [review]

File: src/UcrmPluginSdk/Service/UcrmApi.php (hunk 2 above)
Comment (ajoah) [reply src/UcrmPluginSdk/Service/UcrmApi.php]: no problem ;)

**Review by keksa — APPROVED** [review]

**Comment by keksa** [comment]
Thank you @ajoah!

**Comment by keksa** [comment]
@ajoah One last thing, please add a new test covering the change, take a look at the previous one: https://github.com/Ubiquiti-App/UCRM-Plugin-SDK/blob/3fa4954849f89c7fce935d867a67dd69686bead2/tests/UcrmPluginSdk/Service/UcrmApiTest.php#L146

**Comment by ajoah** [comment]
I am a beginner with PHPUnit and i did not know mock and coverage. I hope i am good 😳

**Comment by keksa** [comment]
@ajoah Thank you, looks fine, I'll probably look into the test after this is merged and improve it a bit, I'll mention you in the PR so you can take a look if you're interested.

**Comment by ajoah** [comment]
Great, thank you :)

**Comment by keksa** [comment]
Thank you @ajoah! I improved the test a bit, you can take a look at https://github.com/Ubiquiti-App/UCRM-Plugin-SDK/pull/12/files

**Pull request merged**